src/Symfony/Component/Security/Http/Controller/UserValueResolver.php

```diff
@@ -35,7 +35,8 @@ public function __construct(TokenStorageInterface $tokenStorage)
     public function supports(Request $request, ArgumentMetadata $argument)
     {
         // only security user implementations are supported
-        if (UserInterface::class !== $argument->getType()) {
+        $type = $argument->getType();
+        if (UserInterface::class !== $type && !is_subclass_of($type, UserInterface::class)) {
```

Comment:
I wonder if l48 should be changed to `return $user instanceof $type` to prevent a type error?